pkg/framework/aries/framework.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package aries
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/context"
+	errors "golang.org/x/xerrors"
+)
+
+// Aries provides access to clients being managed by the framework.
+type Aries struct {
+	transport api.TransportProviderFactory
+}
+
+// Option configures the framework.
+type Option func(opts *Aries) error
+
+// New initializes the Aries framework based on the set of options provided.
+func New(opts ...Option) (*Aries, error) {
+	// get the default framework options
+	defOpts := defFrameworkOpts()
+
+	frameworkOpts := &Aries{}
+
+	// generate framework configs from options
+	for _, option := range append(defOpts, opts...) {
```

Comment:
defOpts has default and append opts to it may duplicate options, this can lead to unexpected behaviour

Reply:
The second opts(user opts) will overwrite the defOpts.

Reply:
Yes - second opt can override.

The other way around this type of problem seems to be to force framework users to pass the default option, rather than implicitly include as the first option. This has its own negative trade-off, of course.

Reply:
Perhaps we could also introduce a function argument like WithoutDefaults()